vllm/model_executor/layers/quantization/qqq.py

```diff
@@ -0,0 +1,279 @@
+from typing import Any, Dict, List, Optional
+
+import torch
+from torch.nn.parameter import Parameter
+
+from vllm import _custom_ops as ops
+from vllm.logger import init_logger
+from vllm.model_executor.layers.linear import LinearBase, LinearMethodBase
+from vllm.model_executor.layers.quantization.base_config import (
+    QuantizationConfig)
+from vllm.model_executor.utils import set_weight_attrs
+
+logger = init_logger(__name__)
+
+MARLIN_QQQ_TILE = 16
+MARLIN_QQQ_MIN_THREAD_N = 64
+MARLIN_QQQ_MIN_THREAD_K = 128
+MARLIN_QQQ_MAX_PARALLEL = 16
+
+MARLIN_QQQ_SUPPORTED_NUM_BITS = [4]
+MARLIN_QQQ_SUPPORTED_GROUP_SIZES = [-1, 128]
+MARLIN_SUPPORTED_SYM = [True]
```

Comment:
This isn't used anywhere. Should it be removed?

Reply:
ok

Reply:
@tlrmchlsmth They are used in [vllm/model_executor/layers/quantization/qqq.py](https://github.com/vllm-project/vllm/pull/5218/commits/f8fba5ea74bfda8c99d904192c3ede3543377025#diff-bb4b60166935dd924ca19794d6af2c83cc115a5d0bbc0bb4b53ad9fdf5b54150R38-R49) and [tests/kernels/test_marlin_gemm.py](https://github.com/vllm-project/vllm/pull/5218/commits/f8fba5ea74bfda8c99d904192c3ede3543377025#diff-7cc4a2d906e22a9eace8c90f99dec52769bfc5a884ef22558f5ace310ed929a9R229-R230). `vllm/model_executor/layers/quantization/gptq_marlin.py` did the similar thing. I think it should be retained.